inc/sitemaps/class-post-type-sitemap-provider.php

```diff
@@ -384,18 +384,22 @@ protected function get_first_links( $post_type ) {
 		$links       = array();
 		$archive_url = false;
 
-		if ( $post_type === 'page' && ! $this->get_page_on_front_id() ) {
+		if ( $post_type === 'page' ) {
 
-			$links[] = array(
-				'loc' => $this->get_home_url(),
+			$front_page = array( 'loc' => $this->get_home_url() );
 
-				// Deprecated, kept for backwards data compat. R.
-				'chf' => 'daily',
-				'pri' => 1,
-			);
-		}
+			$post = $this->get_page_on_front_id() ? get_post( $this->get_page_on_front_id() ) : null;
+			if ( $post ) {
```

Comment:
I would change this to:
```suggestion
			if ( $this->get_page_on_front_id() ) {
                $front_page = $this->get_url( 
			        get_post( $this->get_page_on_front_id() ) 
			    );
```

In this case you only have the `$post` object when it is needed.